### ISSUE TITLE
Is hardOps GPL licensed?

### ISSUE THREAD

**Issue by unfa** [mx1001/hops_p#1]
This version of Hard Ops seems to be very old and incompatible with Blender 2.8 or newer.
May I kindly ask for an update?

I know this is a paid add-on and the sales push the development forward, though it's also GPL-licensed (according to https://blendermarket.com/products/hardopsofficial) so I guess there should be some (legal) way of getting the source code.

**Comment by mx1001** [comment]
https://hardops-manual.readthedocs.io/en/latest/faq/#how-do-i-update-hard-ops-boxcutter

**Comment by unfa** [comment]
Thanks, that's not what I'm looking for :)

I'd gladly donate to the project if it was possible to get it for free (since it's GPL-licensed - or is it?) - I'm teaching audio production on Linux with libre software and my philosophy is to make sure all the tools I use are freely available to everyone (and open-source - that's even more important).

It's not really about the money, it's about making sure tools are accessible to people who can't spend even $1, but can spend extra time to get tools compiled from source etc.

**So are your Blender add-ons licensed under Gnu General Public License or not?**

If they're not - I have no issue with that - and I believe developers deserve to be paid for their work (that's why I donate to FOSS projects I find valuable). I'm just confused - if HardOps and BoxCutter are licensed under GPL, then (IIUC) the license says that [if the software is made public] - the source code must be provided on demand.

I apologize if I'm a bit pushy about this - if the license is not GPL, please correct the product page to fix that, otherwise people will expect source code to be publicly available.

**Comment by unfa** [comment]
Ok, it seems that any add-ons that use Blender API need to have their script (source code) licensed under GPL, as technically they are derivative work, and GPL requires that. That does not include any assets (example files, images, textures, other other non-executable data).

Source: https://blender.stackexchange.com/a/72099

**Comment by oRazeD** [comment]
I wouldn't count on it, Unfa. Some add-on developers tend to avoid sharing source code online... Although purchasing the add-on technically does give you the source code itself.

**Comment by unfa** [comment]
For all I know it is in violation of the software license. It shouldn't be like that.
I understand that the developers don't want to loose sales, but they must have known what they are getting themselves into when they went into this business, right?

**Comment by pati08** [comment]
Yes. This is the exact thing that GPL was designed to prevent. If the addons are considered derivatives under the GPL license, it is a blatant violation of that license not to make the code clearly available. To cite the license:
"You must make sure that they, too, receive
or can get the source code.  And you must show them these terms so they
know their rights."

**Comment by pati08** [comment]
:) https://github.com/AddonsDev/Blender-Add-On

**Comment by unfa** [comment]
I have read up, talked to people and learned that GPL actually allows this.

The requirement to share the source code only applies to distributing the software itself.
If the software can be only obtained by purchase, the source code must come with that purchase,  but that's it. Source doesn't have to be public, unless the software is public.

So I was mistaken, GPL plug-in devs are not obligated to release the source code to the plug-ins for free.

**Comment by oRazeD** [comment]
Any Blender add-ons inherit Blender's licensing, meaning yes, you are supposed to make the source available for free somewhere if you are also selling it as a product. Blender Market; which doesn't allow you to create free products, is a particularly large example of something the blender devs have been fighting for a long time.

**Comment by pati08** [comment]
I deleted my prior comment because I looked at the dates and thought I had commented on a dead discussion. It basically said that you not only have to make them available, but also make the rights apparent to the users.

Also, see this: https://www.blender.org/about/license/
It says that the source has to be just as free as Blender as it uses the Blender Python API, an integral part of Blender. So, @unfa it explicitly says the opposite on the Blender licensing page.

LMK If I'm wrong.
